### ISSUE TITLE
🐞 Bug – elm generate pipeline does not know how to parse Elm compiler error

### ISSUE THREAD

**Issue by dz0ny** [elm-land/elm-land#137]
## Problem

The ```elm generate``` pipeline does not know how to parse Elm compiler error.

```js
/Users/dz0ny/rnk/frontend/node_modules/elm-land/src/effects.js:31
    throw err

RangeError: Invalid count value
    at String.repeat (<anonymous>)
    at header (/Users/dz0ny/rnk/frontend/node_modules/elm-land/src/vite-plugins/elm/elm-error-json.js:169:60)
    at toColoredTerminalOutput (/Users/dz0ny/rnk/frontend/node_modules/elm-land/src/vite-plugins/elm/elm-error-json.js:224:30)
    at Array.map (<anonymous>)
    at generateElmFiles (/Users/dz0ny/rnk/frontend/node_modules/elm-land/src/effects.js:291:16)
    at runNextTicks (node:internal/process/task_queues:60:5)
    at listOnTimeout (node:internal/timers:538:9)
    at process.processTimers (node:internal/timers:512:7)
    at async generate (/Users/dz0ny/rnk/frontend/node_modules/elm-land/src/effects.js:408:3)
    at async Object.run (/Users/dz0ny/rnk/frontend/node_modules/elm-land/src/effects.js:615:22)
```

JS dump
```js
error {
  type: 'error',
  path: '/Users/dz0ny/rnk/frontend/src/Pages/Reports/Reports_id_/Keyword/Keyword_id_/Page/Page_id_/Body.elm',
  title: 'UNEXPECTED TYPE ANNOTATION',
  message: [
    'I found an unexpected type annotation on this ',
    { bold: false, underline: false, color: 'YELLOW', string: 'page' },
    ' function.\n\n',
    '  page : ',
    'Shared.Model -> PageRoute -> Page Model Msg',
    '\n         ',
    {
      bold: false,
      underline: false,
      color: 'RED',
      string: '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^'
    },
    '\nI recommend one of these annotations:\n\n',
    '  page : ',
    {
      bold: false,
      underline: false,
      color: 'YELLOW',
      string: '{ reports_id : String, keyword_id : String, page_id : String } -> View msg\n' +
        '\n'
    },
    '  page : ',
    {
      bold: false,
      underline: false,
      color: 'YELLOW',
      string: 'Page Model Msg\n\n'
    },
    '  page : ',
    {
      bold: false,
      underline: false,
      color: 'YELLOW',
      string: 'Shared.Model -> Route { reports_id : String, keyword_id : String, page_id : String } -> Page Model Msg\n' +
        '\n'
    },
    '  page : ',
    {
      bold: false,
      underline: false,
      color: 'YELLOW',
      string: 'Auth.User -> Shared.Model -> Route { reports_id : String, keyword_id : String, page_id : String } -> Page Model Msg\n' +
        '\n'
    },
    'Although Elm annotations are optional, Elm Land requires an annotation for\n' +
      'this particular function to avoid showing errors in generated code.\n' +
      '\n',
    { bold: false, underline: true, color: null, string: 'Hint:' },
    ' Read https://elm.land/problems#unexpected-type-annotation to learn more'
  ]
}
problem {
  type: 'error',
  path: '/Users/dz0ny/rnk/frontend/src/Pages/Reports/Reports_id_/Keyword/Keyword_id_/Page/Page_id_/Body.elm',
  title: 'UNEXPECTED TYPE ANNOTATION',
  message: [
    'I found an unexpected type annotation on this ',
    { bold: false, underline: false, color: 'YELLOW', string: 'page' },
    ' function.\n\n',
    '  page : ',
    'Shared.Model -> PageRoute -> Page Model Msg',
    '\n         ',
    {
      bold: false,
      underline: false,
      color: 'RED',
      string: '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^'
    },
    '\nI recommend one of these annotations:\n\n',
    '  page : ',
    {
      bold: false,
      underline: false,
      color: 'YELLOW',
      string: '{ reports_id : String, keyword_id : String, page_id : String } -> View msg\n' +
        '\n'
    },
    '  page : ',
    {
      bold: false,
      underline: false,
      color: 'YELLOW',
      string: 'Page Model Msg\n\n'
    },
    '  page : ',
    {
      bold: false,
      underline: false,
      color: 'YELLOW',
      string: 'Shared.Model -> Route { reports_id : String, keyword_id : String, page_id : String } -> Page Model Msg\n' +
        '\n'
    },
    '  page : ',
    {
      bold: false,
      underline: false,
      color: 'YELLOW',
      string: 'Auth.User -> Shared.Model -> Route { reports_id : String, keyword_id : String, page_id : String } -> Page Model Msg\n' +
        '\n'
    },
    'Although Elm annotations are optional, Elm Land requires an annotation for\n' +
      'this particular function to avoid showing errors in generated code.\n' +
      '\n',
    { bold: false, underline: true, color: null, string: 'Hint:' },
    ' Read https://elm.land/problems#unexpected-type-annotation to learn more'
  ]
}
```

## Your setup

elm-land v0.19.3

**Comment by ryannhg** [comment]
That's strange, I would have expected this line to properly format the validation error in the console:
https://github.com/elm-land/elm-land/blob/main/projects/cli/src/effects.js#L291 

It looks like it's hitting the `uncaughtException` line, which usually happens when something isn't awaited, returned, or try/catch-ed correctly.
https://github.com/elm-land/elm-land/blob/main/projects/cli/src/effects.js#L31

@dz0ny – does this work when you run `elm-land build` or is there something broken on both commands?

**Comment by dz0ny** [comment]
Happens for elm-land-server as well and it gets propagated to the browser view.

**Comment by ryan-haskell** [comment]
Good news– I just released [📦 Elm Land v0.20.0](https://github.com/elm-land/elm-land/releases/tag/v0.20.0) which resolves this issue (more details in the release linked above)

Thanks for improving Elm Land! 🌈
